integration_tests/src/main/python/regexp_test.py

```diff
@@ -427,21 +427,23 @@ def test_regexp_extract():
                 'regexp_extract(a, "([0-9]+)", 1)',
                 'regexp_extract(a, "([0-9])([abcd]+)", 1)',
                 'regexp_extract(a, "([0-9])([abcd]+)", 2)',
-                'regexp_extract(a, "^([a-d]*)([0-9]*)([a-d]*)\\z", 1)',
```

Comment:
Is there a reason we need to remove the tests with `\\z`?

Reply:
From what I gather `\z` is disabled:

```
        case 'z' =>
          // cuDF does not support "\z" except for in split mode
          throw new RegexUnsupportedException(
            "\\z is not supported on GPU for find or replace",
            regex.position)
```

Also, this test wasn't really testing that, in PySpark, for escaping, you need to double escape. Basically it's a string inside a SQL command that needs a `\\z`. So the Python string equivalent would be `\\\\z`.